src/include/codegen/util/sorter.h

```diff
@@ -143,6 +219,17 @@ class Sorter {
    */
   void TransferMemoryBlocks(Sorter &target);
 
+  /**
+   * Build a (max) heap from the tuples currently stored in the sorter instance.
+   */
+  void BuildHeap();
+
+  /**
+   * Sift down the element at the root of the heap while maintaining the heap
+   * property.
+   */
+  void HeapSiftDown();
```

Comment:
What is the difference of this function and STL `push_heap()`?

Reply:
In essence, they are the same. But, `push_heap` alone doesn't give us the functionality we need. We need to remove the smallest (largest) element off the min (max) heap and insert a new element. We could use a pair of `push_heap` and `pop_heap` calls, but we've managed to achieve the same functionality with a single `O(logn)` operation.